pkg/kubelet/cm/dra/state/checkpoint.go

```diff
@@ -63,6 +105,37 @@ func (dc *DRAManagerCheckpoint) VerifyChecksum() error {
 	ck := dc.Checksum
 	dc.Checksum = 0
 	err := ck.Verify(dc)
+	if err == errors.ErrCorruptCheckpoint {
+		// Verify with old structs without ResourceHandles field
+		// TODO: remove in Beta
+		entries := ClaimInfoStateListWithoutResourceHandles{}
+		for _, entry := range dc.Entries {
+			entries = append(entries, ClaimInfoStateWithoutResourceHandles{
+				DriverName: entry.DriverName,
+				ClassName:  entry.ClassName,
+				ClaimUID:   entry.ClaimUID,
+				ClaimName:  entry.ClaimName,
+				Namespace:  entry.Namespace,
+				PodUIDs:    entry.PodUIDs,
+				CDIDevices: entry.CDIDevices,
+			})
+		}
+		oldcheckpoint := &DRAManagerCheckpointWithoutResourceHandles{
+			Version:  checkpointVersion,
+			Entries:  entries,
+			Checksum: 0,
+		}
+		// Calculate checksum for old checkpoint
+		object := dump.ForHash(oldcheckpoint)
+		object = strings.Replace(object, "DRAManagerCheckpointWithoutResourceHandles", "DRAManagerCheckpoint", 1)
+		object = strings.Replace(object, "ClaimInfoStateListWithoutResourceHandles", "ClaimInfoStateList", 1)
+		hash := fnv.New32a()
+		fmt.Fprintf(hash, "%v", object)
+		err = nil
+		if ck != checksum.Checksum(hash.Sum32()) {
+			err = errors.ErrCorruptCheckpoint
+		}
```

Comment:
Can you put this in a helper function to keep VerifyChecksum call shorter.

Reply:
done